packages/react/tooltip/src/Tooltip.tsx

```diff
@@ -157,13 +158,15 @@ const Tooltip: React.FC<TooltipProps> = (props: ScopedProps<TooltipProps>) => {
     prop: openProp,
     defaultProp: defaultOpen,
     onChange: (open) => {
-      if (open) {
-        // we dispatch here so `TooltipProvider` isn't required to
-        // ensure other tooltips are aware of this one opening.
-        document.dispatchEvent(new CustomEvent(TOOLTIP_OPEN));
-        onOpen();
-      }
-      onOpenChange?.(open);
+      ReactDOM.flushSync(() => {
```

Comment:
In here, I've decided to wrap the whole `onChange` handler to "batch" all updates scheduled by within this callback